src/op_analytics/datapipeline/models/compute/udfs.py

```diff
@@ -34,7 +37,23 @@ def create_duckdb_macros(duckdb_client: duckdb.DuckDBPyConnection):
     
     --Get the length in bytes for binary data that is encoded as a hex string
     CREATE OR REPLACE MACRO hexstr_bytelen(x)
-    AS (length(x) - 2) / 2 
+    AS (length(x) - 2) / 2;
+                      
+    --Count non-zero bytes for binary data that is encoded as a hex string. We don't use hexstr_bytelen because we need to substring the input data.
+    CREATE OR REPLACE MACRO hexstr_nonzero_bytes(x)
+    AS length( REPLACE(TO_HEX(FROM_HEX(SUBSTR(x, 3))), '00', '') ) / 2;
+    
+    --Count non-zero bytes for binary data that is encoded as a hex string
+    CREATE OR REPLACE MACRO hexstr_zero_bytes(x)
+    AS hexstr_bytelen(x) - hexstr_nonzero_bytes(x);
+    
+    --Calculate calldata gas used for binary data that is encoded as a hex string (can be updated by an EIP)
+    CREATE OR REPLACE MACRO hexstr_calldata_gas(x)
+    AS 16*hexstr_nonzero_bytes(x) + 4*hexstr_zero_bytes(x);
+    
+    --Get the method id for input data. This is the first 4 bytes, or first 10 string characters for binary data that is encoded as a hex string.
+    CREATE OR REPLACE MACRO hexstr_method_id(x)
+    AS substring(x,1,10);
     """)
```

Comment:
Is there some way that we could or should pass in the alias for each table? If not, then it ~should be some standard to always call transactions "t", blocks "b", logs "l", traces "r", etc